base/compiler/abstractinterpretation.jl

```diff
@@ -860,17 +860,15 @@ function abstract_iteration(interp::AbstractInterpreter, @nospecialize(itft), @n
     end
     # From here on, we start asking for results on the widened types, rather than
     # the precise (potentially const) state type
-    statetype = widenconst(statetype)
     valtype = widenconst(valtype)
```

Comment:
```suggestion
    valtype = Bottom
```

Reply:
True; I was basically `tmerge`ing this with itself after the first iteration, so could indeed just start with `Bottom` here to maybe save a bit. But more importantly, that will ensure we don't declare a fixpoint in the first iteration, so we could go back to widening the state here and using `<:` in the loops break condition. Let me play with that locally and see whether I missed anything in that reasoning.

Reply:
`valtype` is the type of the previous element, not the type of the current element. It was incorrect (though conservative) to include it

Reply:
Ah, true.